src/EditorFeatures/Core/Implementation/InlineRename/InlineRenameSession.OpenTextBufferManager.cs

```diff
@@ -227,24 +226,23 @@ private void OnTextBufferChanged(object sender, TextContentChangedEventArgs args
                     var trackingSpansAfterEdit = new NormalizedSpanCollection(GetEditableSpansForSnapshot(args.After).Select(ss => (Span)ss));
                     var spansTouchedInEdit = new NormalizedSpanCollection(args.Changes.Select(c => c.NewSpan));
 
-                    var intersection = NormalizedSpanCollection.Intersection(trackingSpansAfterEdit, spansTouchedInEdit);
-
-                    if (intersection.Count == 0)
+                    var intersectionSpans = NormalizedSpanCollection.Intersection(trackingSpansAfterEdit, spansTouchedInEdit);
+                    if (intersectionSpans.Count == 0)
                     {
                         // In Razor we sometimes get formatting changes during inline rename that
                         // do not intersect with any of our spans. Ideally this shouldn't happen at
                         // all, but if it does happen we can just ignore it.
                         return;
                     }
-                    else if (intersection.Count > 1)
-                    {
-                        Contract.Fail("we can't allow edits to touch multiple spans");
-                    }
 
-                    var intersectionSpan = intersection.Single();
-                    var singleTrackingSpanTouched = GetEditableSpansForSnapshot(args.After).Single(ss => ss.IntersectsWith(intersectionSpan));
-                    _activeSpan = _referenceSpanToLinkedRenameSpanMap.Where(kvp => kvp.Value.TrackingSpan.GetSpan(args.After).Contains(intersectionSpan)).Single().Key;
+                    // Cases with invalid identifiers may cause there to be multiple intersection
+                    // spans, but they should still all map to a single tracked rename span.
```

Comment:
Yeah, it requires *really* invalid identifiers. Something like renaming ```two``` to ```one two three``` which may be interpreted as two changes (the addition of "one" and the addition of "three").